### PR TITLE
Rails upgrade 6 or 7

### PR DESCRIPTION
**Pull request by mestachs**
## Self proof reading checklist

- [ ] Did I run Pronto and fixed all warnings (pronto run -c origin/dev)
- [ ] Make sure all naming and code will remain understandable in 6 month by someone new to the project or by you.
- [ ] Did I test the right thing?
- [ ] Did I test corner cases, non happy path (defensive testing)?
- [ ] Check that I used the ad-hoc patterns and created files accordingly (Presenters, Services, Search object, Form object,...)
- [ ] Check code efficiency with record intensive project
- [ ] Update documentation,readme accordingly

Thanks!

### CODE REVIEW

File: app/models/project.rb [hunk 1]
```diff
@@ -394,7 +394,7 @@ def changelog(other_project = original)
 
     diff_symbols = { "+" => :added, "-" => :removed, "~" => :modified }
     all_names = to_unified_names.merge(other_project.to_unified_names)
-    HashDiff.diff(other_project.to_unified_h, to_unified_h).map do |hash_diff|
+    Hashdiff.diff(other_project.to_unified_h, to_unified_h).map do |hash_diff|
```
Comment (pjaspers) [inline app/models/project.rb:397]: Intentional?

File: app/models/rule_types/base_rule_type.rb [hunk 2]
```diff
@@ -75,8 +76,8 @@ def package_formula_uniqness
       formula_by_codes.each do |code, formulas|
         next unless formulas.size > 1
 
-        rule.errors[:formulas] << "Formula's code must be unique,"\
-          " you have #{formulas.size} formulas with '#{code}'"
+        rule.errors.add(:formulas, "Formula's code must be unique,"\
```
Comment (pjaspers) [inline app/models/rule_types/base_rule_type.rb:79]: I'll miss the `<<` for errors

File: app/serializers/invoicing_job_serializer.rb [hunk 3]
```diff
@@ -32,12 +32,12 @@
 class InvoicingJobSerializer
   include FastJsonapi::ObjectSerializer
   set_key_transform :camel_lower
-  attribute :org_unit, &:orgunit_ref
+  attribute :org_unit do |rec| rec.orgunit_ref end
```
Comment (pjaspers) [inline app/serializers/invoicing_job_serializer.rb:35]: I would have gone `{ }` for the single line thing, but this words as well, they don't allow a proc anymore?

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -3,6 +3,8 @@
   <head>
     <title>Open RBF 2.0</title>
     <%= csrf_meta_tags %>
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.15.4/css/solid.css" integrity="sha384-Tv5i09RULyHKMwX0E8wJUqSOaXlyu3SQxORObAI08iUwIalMmN5L6AvlPX2LMoSE" crossorigin="anonymous"/>
```
Comment (pjaspers) [inline app/views/layouts/application.html.erb:6]: Might be worth it to just download it, and serve it ourselves.

File: config/initializers/rails_admin.rb [hunk 5]
```diff
@@ -3,6 +3,7 @@
 end
 
 RailsAdmin.config do |config|
+  config.asset_source = :sprockets
```
Comment (pjaspers) [inline config/initializers/rails_admin.rb:6]: This feels like a line that looks simple but hides a lot of hair pulling.

**Review by pjaspers — APPROVED** [review]
Scorpio moving to another Rails version!

Looks good to me

File: app/models/project.rb (hunk 1 above)
Comment (mestachs) [reply app/models/project.rb:397]: it's weird yes, as if the gem changed it afterwards 
https://github.com/liufengyun/hashdiff#diff or it's another gem ?

**Comment by pjaspers** [comment]
https://github.com/liufengyun/hashdiff/commit/3a8ce951455cc6c3e42b6ce6587b02d16cbdbed9

So looks to be intentional :)
